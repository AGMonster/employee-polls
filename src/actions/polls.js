export const RECIEVE_POLLS = 'RECIEVE_POLLS'
export const NEW_POLL = 'NEW_POLL'
export const ANSWER_POLL = 'ANSWER_POLL'

export function recievePolls(polls) {
    return {
        type: RECIEVE_POLLS,
        polls
    }
}

export function newPoll(poll) {
    return {
        type: NEW_POLL,
        poll
    }
}

export function answerPoll(answerInfo) {
    return {
        type: ANSWER_POLL,
        answerInfo
    }
}
