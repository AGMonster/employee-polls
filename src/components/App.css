.App {
  text-align: center;
}
.App-logo{
  height: 500px;
  object-fit:cover;
  object-position:50% 50%;
}
.nav-items {
  display: flex;
  list-style: none;
  justify-content: flex-start;
  margin-bottom: 0px;
}
.nav-item {
  width : 120px;
}
.nav-item a{
  opacity: 0.65;
  text-decoration: none;
  cursor: pointer;
}
.nav-item .active {
  opacity: 1;
  border-bottom:2px solid teaL;
  padding: 16px;
}
.logout{
  display: flex;
  opacity: 1;
  justify-content: space-evenly;
  align-items: center;
  margin-right: 30px;
}
.logout .loggedin-user{
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  margin-right: 20px;
}
.nav-item.user {
  opacity: 1;
  width : 200px;
  margin-left:auto;
}
.new-poll.sub-heading {
  opacity: 0.35;
}
.profile-info {
  display: flex;
  align-items: center;
}
.user-info {
  display: flex;
  flex-direction: column;
}
.avatar {
  height: 30px;
  border-radius: 15px;
  margin-right: 10px;
}
.divider {
  opacity: 0.25;
  margin-top: 0.5rem;
}
.leaderboard-grid {
  display: flex;
  justify-content: center;
  align-items: center;
}
td.username, td.polls, td.votes {
  width : 40%;
}
table {
  width: 50%;
  border: 1px solid black;
}
td, th {
  border: 1px solid lightgray;
  text-align: left;
  padding: 8px;
}
.poll-list {
  border: 1px solid lightgray;
  margin: 10px;
}
.poll-divider {
  color: lightgray;
  opacity: 0.25;
}
.poll-cards {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  flex-wrap: wrap;
  gap: 16px;
}
.poll-card {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 200px;
  height: 130px;
  border: 1px solid lightgray;
  border-radius: 5px;
  margin: 10px;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  transition: 0.3s;
}
.show-poll-btn {
  width: 160px;
  border: 2px solid teal;
  height: 36px;
  cursor: pointer;
}
.timestamp {
  opacity: 0.4;
  font-size: 14px;
}
.poll-card:hover {
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}
.poll-form {
  display: flex;
  flex-direction: column;
  margin-left: auto;
  margin: auto;
  width : 50%;
}
.poll-form input {
  border-radius: 5px;
  height: 32px;
  margin: 16px;
  border: 1px solid lightgray;
}
.poll-form button {
  border-radius: 5px;
  height: 40px;
  width: 300px;
  margin: 16px auto;
  border: 1px solid lightgray;
  cursor: pointer;
}
.poll-detail {
  display: flex;
  justify-content: space-around;
  width: 65%;
  margin-left: auto;
  margin-right: auto;
}
.poll-author-avatar {
  height: 150px;
  border-radius: 50px;
}
.option button, .option .option-text {
  width: 350px;
  border: 10px lightgray;
  background-color: lightgray;
  margin-right: 10px;
  padding:16px;
  box-sizing: border-box;
}
.option-text.selected {
  border: 3px solid darkseagreen;
}
.option button {
  cursor: pointer;
  background-color: teal;
}
.login-form {
  display: flex;
  flex-direction: column;
  width: 40%;
  height: 100%;
  justify-content: center;
  margin-left: auto;
  margin-right: auto;
}
.user-select {
  height: 48px;
  margin: 20px;
  border-radius: 8px;
}
.login-btn {
  width: 160px;
  height: 48px;
  margin: 20px auto;
  border-radius: 8px;
  cursor: pointer;
}