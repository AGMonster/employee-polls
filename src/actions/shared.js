import { showLoading, hideLoading } from 'react-redux-loading-bar' 
import { recievePolls, newPoll, answerPoll } from "./polls"
import { recieveUsers } from "./users"
import { setAuthedUser } from "./authedUser"
import { getInitialData, savePoll, savePollAnswer, getusers } from "../utils/api"

export function handleSetAuthedUser(authedUser) {
    return (dispatch) => {
        dispatch(showLoading());
        getusers().then((users ) => {
            let user = users[authedUser]
            if(user) {
                dispatch(setAuthedUser(authedUser));
            }
            else {
                throw Error('User Not Found');
            }
        }).then(() => {
            dispatch(handleInitialData());
        }).catch((error) => {
            console.log(error);
        });
    }
}

export function handleInitialData() {
    return (dispatch) => {
        getInitialData().then(({users, polls}) => {
            dispatch(recievePolls(polls))
            dispatch(recieveUsers(users))
        }).then(() => {
            dispatch(hideLoading())
        }).catch((error) => {
            console.log(error);
        });
    }
}

export function handleNewPoll(optionOneText, optionTwoText) {
    return (dispatch, getState) => {
        dispatch(showLoading()); 
        let {authedUser} =  getState()
        let pollInfo = { optionOneText, optionTwoText, author : authedUser }
        savePoll(pollInfo).then((poll) => {
            dispatch(newPoll(poll));
        }).then(()=>{
            dispatch(hideLoading());
        }).catch((error) => {
            console.log(error);
        });
    }
}

export function handleAnswerPoll(option, pollId) {
    return (dispatch, getState) => {
        let { authedUser } = getState()
        let answerInfo = { 
            authedUser: authedUser,
            qid: pollId,
            answer: option 
        };
        savePollAnswer(answerInfo).then((res) => {
            dispatch(answerPoll(answerInfo))
        }).catch((error) => {
            console.log(error);
        });
    }
}