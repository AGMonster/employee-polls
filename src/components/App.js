import '../App.css';
import Nav from './Nav'
import Dashboard from './Dashboard';
import { connect } from 'react-redux'
import { Fragment } from 'react';
import { Routes, Route } from 'react-router-dom'
import NewPoll from './NewPoll';
import Leaderboard from './Leaderboard';
import PollDetail from './PollDetail';
import Login from './Login';
import NotFound from './NotFound';
import LoadingBar from 'react-redux-loading-bar'

function App({IsLoggedIn, isLoading}) {
  return (
    <Fragment>
      <LoadingBar/>
      { 
        !IsLoggedIn ? <Login/> :
        <div className="App">
          <header className="header">
            <Nav/>
          </header>
          { 
            !isLoading &&
            <Routes>
              <Route path="/" exact element={<Dashboard/>}></Route>
              <Route path="/add" element={<NewPoll/>}></Route>
              <Route path="/leaderboard" element={<Leaderboard/>}></Route>
              <Route path="/questions/:question_id" element={<PollDetail/>}></Route>
              <Route path="*" element={<NotFound />} />
            </Routes> 
          }
        </div>
    }
    </Fragment>
  );
}

const mapStateToProps = ({authedUser, loadingBar}) => {
  return {
    IsLoggedIn : authedUser != null,
    isLoading: loadingBar.default === 1
  }
}

export default connect(mapStateToProps)(App);
