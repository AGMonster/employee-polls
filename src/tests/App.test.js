import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux';
import reducer from '../reducers'
import { applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { MemoryRouter } from 'react-router';
import App from '../components/App';
import {  handleSetAuthedUser } from '../actions/shared';

  test('App component matches snapshot', async() => {
    const store = createStore(reducer, applyMiddleware(thunk))
    await store.dispatch(handleSetAuthedUser('sarahedo'));
    await new Promise((res) => setTimeout(() => res(), 2200));

    let component = render(
        <MemoryRouter>
        <Provider store = {store}>
          <App />
        </Provider>
        </MemoryRouter>   );
    expect(component).toMatchSnapshot()
 })
