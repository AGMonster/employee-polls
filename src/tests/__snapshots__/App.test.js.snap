// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        class="App"
      >
        <header
          class="header"
        >
          <nav>
            <ul
              class="nav-items"
            >
              <li
                class="nav-item"
              >
                 
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                   Dashboard
                </a>
                 
              </li>
              <li
                class="nav-item"
              >
                 
                <a
                  class=""
                  href="/leaderboard"
                >
                   Leaderboard 
                </a>
              </li>
              <li
                class="nav-item"
              >
                 
                <a
                  class=""
                  href="/add"
                >
                   New
                </a>
                 
              </li>
              <li
                class="nav-item user"
              >
                 
                <div
                  class="logout"
                >
                  <div
                    class="loggedin-user"
                  >
                    <img
                      alt="avatar of the user"
                      class="avatar"
                      src="https://tylermcginnis.com/would-you-rather/sarah.jpg"
                    />
                    <span
                      class="user"
                    >
                       
                      sarahedo
                    </span>
                  </div>
                  <a
                    href="/"
                  >
                    Logout
                  </a>
                </div>
                 
              </li>
            </ul>
            <hr
              class="divider"
            />
          </nav>
        </header>
        <div>
          <button
            class="toggle-view-btn"
          >
             Toggle Active View 
          </button>
          <div
            class="poll-list"
          >
            <p>
              <strong>
                New Questions
              </strong>
            </p>
            <hr
              class="poll-divider"
            />
            <div
              class="poll-cards"
            >
              <div
                class="poll-card"
              >
                <span>
                   
                  johndoe
                </span>
                <span
                  class="timestamp"
                >
                   
                  12:46AM | 1/5/2017
                </span>
                <a
                  href="/questions/xj352vofupe1dqz9emx13r"
                >
                  <button
                    class="show-poll-btn"
                    data-testid="show-btn"
                  >
                     show 
                  </button>
                </a>
              </div>
              <div
                class="poll-card"
              >
                <span>
                   
                  tylermcginnis
                </span>
                <span
                  class="timestamp"
                >
                   
                  5:39:PM | 15/3/2017
                </span>
                <a
                  href="/questions/vthrdm985a262al8qx3do"
                >
                  <button
                    class="show-poll-btn"
                    data-testid="show-btn"
                  >
                     show 
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      class="App"
    >
      <header
        class="header"
      >
        <nav>
          <ul
            class="nav-items"
          >
            <li
              class="nav-item"
            >
               
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                 Dashboard
              </a>
               
            </li>
            <li
              class="nav-item"
            >
               
              <a
                class=""
                href="/leaderboard"
              >
                 Leaderboard 
              </a>
            </li>
            <li
              class="nav-item"
            >
               
              <a
                class=""
                href="/add"
              >
                 New
              </a>
               
            </li>
            <li
              class="nav-item user"
            >
               
              <div
                class="logout"
              >
                <div
                  class="loggedin-user"
                >
                  <img
                    alt="avatar of the user"
                    class="avatar"
                    src="https://tylermcginnis.com/would-you-rather/sarah.jpg"
                  />
                  <span
                    class="user"
                  >
                     
                    sarahedo
                  </span>
                </div>
                <a
                  href="/"
                >
                  Logout
                </a>
              </div>
               
            </li>
          </ul>
          <hr
            class="divider"
          />
        </nav>
      </header>
      <div>
        <button
          class="toggle-view-btn"
        >
           Toggle Active View 
        </button>
        <div
          class="poll-list"
        >
          <p>
            <strong>
              New Questions
            </strong>
          </p>
          <hr
            class="poll-divider"
          />
          <div
            class="poll-cards"
          >
            <div
              class="poll-card"
            >
              <span>
                 
                johndoe
              </span>
              <span
                class="timestamp"
              >
                 
                12:46AM | 1/5/2017
              </span>
              <a
                href="/questions/xj352vofupe1dqz9emx13r"
              >
                <button
                  class="show-poll-btn"
                  data-testid="show-btn"
                >
                   show 
                </button>
              </a>
            </div>
            <div
              class="poll-card"
            >
              <span>
                 
                tylermcginnis
              </span>
              <span
                class="timestamp"
              >
                 
                5:39:PM | 15/3/2017
              </span>
              <a
                href="/questions/vthrdm985a262al8qx3do"
              >
                <button
                  class="show-poll-btn"
                  data-testid="show-btn"
                >
                   show 
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
